jaeger_client/constants.py

```diff
@@ -33,10 +33,10 @@
 DEFAULT_FLUSH_INTERVAL = 1
 
 # Name of the HTTP header used to encode trace ID
-TRACE_ID_HEADER = b'uber-trace-id'
```

Comment:
I believe in Python 2.7 string and bytes types are essentially equivalent:

```
Python 2.7.12 (default, Oct 11 2016, 05:20:59)

>>> a = 'helloпривет'
>>> b = b'helloпривет'
>>> a
'hello\xd0\xbf\xd1\x80\xd0\xb8\xd0\xb2\xd0\xb5\xd1\x82'
>>> b
'hello\xd0\xbf\xd1\x80\xd0\xb8\xd0\xb2\xd0\xb5\xd1\x82'
>>> type(a)
<type 'str'>
>>> type(b)
<type 'str'>
```

They are different in Python 3 though:
```
>>> a='helloпривет'
>>> b=b'helloпривет'
  File "<stdin>", line 1
SyntaxError: bytes can only contain ASCII literal characters.
>>> b=b'hello'
>>> a
'helloпривет'
>>> b
b'hello'
>>> type(a)
<class 'str'>
>>> type(b)
<class 'bytes'>
```

So, with this string being explicitly marked as bytes, I had the following error when running tests in Python3:

```
tests/test_tracer.py:178: in test_tracer_tags_hostname
    t = Tracer(service_name='x', reporter=reporter, sampler=sampler)
jaeger_client/tracer.py:62: in __init__
    debug_id_header=debug_id_header,
jaeger_client/codecs.py:56: in __init__
    self.trace_id_header = trace_id_header.lower().replace('_', '-')
E   TypeError: a bytes-like object is required, not 'str'
```

Which makes sense, because `'_'` is a string object, not bytes in Python 3. Changing `trace_id_header.lower().replace('_', '-')` to `trace_id_header.lower().replace(b'_', b'-')` was fixing this test, but crashing few other ones which were using trace id string without `b` prefix:

```
    def test_context_from_readable_headers(self):
        # provide headers all the way through Config object
        config = Config(
            service_name='test',
            config={
                'trace_id_header': 'Trace_ID',
                'baggage_header_prefix': 'Trace-Attr-',
            })
        tracer = config.create_tracer(
...

tests/test_codecs.py:170: in test_context_from_readable_headers
    sampler=ConstSampler(True),
jaeger_client/config.py:279: in create_tracer
    debug_id_header=self.debug_id_header,
jaeger_client/tracer.py:62: in __init__
    debug_id_header=debug_id_header,
jaeger_client/codecs.py:56: in __init__
    self.trace_id_header = trace_id_header.lower().replace(b'_', b'-')
E   TypeError: replace() argument 1 must be str, not bytes
```

So, weighing the options of changing every affected test and changing just single line `TRACE_ID_HEADER = b'uber-trace-id'` I've chosen the one with the least amount of changes.

Talking about urllib2 issue, I'm not sure what could have caused it given that bytes and strings are same in Py2 – maybe you have an example?